src/lib.rs

```diff
@@ -0,0 +1 @@
+pub mod merkle;
```

Comment:
I think having a directory for the "merkle" module, with a single file inside is too much. Consider having a single file for the `merkle` module at the root instead. If you plan on having more modules in the `merkle` module then it's okay.

Reply:
I usually like having a separate folder from the level of the `main.rs` to separate the code logic from it, but you are right, the `main()` here does not have anything so it's an overkill having a directory with a single file